HopsanGenerator/src/GeneratorTypes.cpp

```diff
@@ -668,7 +669,115 @@ void getInterfaces(QList<InterfacePortSpec> &interfaces, hopsan::ComponentSystem
     }
 }
 
+void getModelVariables(hopsan::ComponentSystem *pSystem, QList<ModelVariableSpecification> &vars, QStringList &systemHierarchy) {
+    std::vector<hopsan::HString> names = pSystem->getSubComponentNames();
+    for(size_t i=0; i<names.size(); ++i)
+    {
+        hopsan::Component *pComponent = pSystem->getSubComponent(names[i]);
+        hopsan::HString typeName = pComponent->getTypeName();
 
+        if(typeName == "SignalInputInterface")
+        {
+            vars.append(ModelVariableSpecification(systemHierarchy, names[i].c_str(), "out", "y", hopsan::NodeSignal::Value, pSystem->getSubComponent(names[i])->getPort("out")->getStartValue(hopsan::NodeSignal::Value), ModelVariableCausality::Input));
+        }
+        else if(typeName == "SignalOutputInterface")
+        {
+            vars.append(ModelVariableSpecification(systemHierarchy, names[i].c_str(), "in", "y", hopsan::NodeSignal::Value, pSystem->getSubComponent(names[i])->getPort("in")->getStartValue(hopsan::NodeSignal::Value), ModelVariableCausality::Output));
+        }
+        else if(typeName == "MechanicInterfaceQ")
+        {
+            vars.append(ModelVariableSpecification(systemHierarchy, names[i].c_str(), "P1", "f", hopsan::NodeMechanic::Force, pSystem->getSubComponent(names[i])->getPort("P1")->getStartValue(hopsan::NodeMechanic::Force), ModelVariableCausality::Input));
```

Comment:
I found a different solution using the `NodeDataDescription` class in HopsanCore. Some hardcoded stuff is still needed for the interface components, but that is innevitable since they are in fact a kind of "custom components".